### ISSUE TITLE
Accessing FA from localhost

### ISSUE THREAD

**Issue by NickPelov** [FusionAuth/fusionauth-typescript-client#39]
I've created a React app that uses the fusionauth-typescript-client to connect to a self hosted FA instance.

The requests are blocked by the CORS policy (when developing locally) and I tried a couple of different setups for the FA CORS policy, but nothing seems to work as expected.

We don't want to create a backend gateway just to bypass the CORS policy on localhost, because the production web-apps are whitelisted and don't have this issue. Is there some sort of configuration we can specify or a particular way to setup/initialize the fusionauth-typescript-client to bypass this?

Thanks a lot for the help!

**Comment by mooreds** [comment]
Hiya,

What browser are you using? It looks like chrome has issues with localhost and CORS: https://stackoverflow.com/questions/10883211/deadly-cors-when-http-localhost-is-the-origin 

That post has some suggestions (chrome extensions, using a service like lvh.me, start up chrome with `--disable-web-security`).

**Comment by nscarlson** [comment]
Using the `cors` package, here are my minimal CORS options currently working for me:

```js
    const corsOptions = {
        origin: FA_URL,
        credentials: true,
    }
```

With express:
```js
    app.use(cors(corsOptions))
```

With Apollo:
```js
    apolloServer.applyMiddleware({
        app,
        cors: corsOptions,
        path: '/',
    })
```

**Comment by NickPelov** [comment]
Sorry for the late response on this our work on the authentication was put on hold for a bit.

I am indeed using chrome so that is probably the issue. But we also tried deploying our app and we get:

> Response to preflight request doesn't pass access control check: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.

Our app is hosted on a domain a (https://example.example.com/login) and our fusion auth instance is running on (https://example2.otherdomain.com).
We couldn't find a way to set a flag or to set an option for the client to use the correct headers. Are we doing something wrong?

Our code looks something along these lines:


```
    const client = new FusionAuthClient(apiKey, serviceUrl, tenantId)

    const login = (username, password) => {
        const obj = {
            loginId: username,
            password: password,
            applicationId: applicationId,
        } 
        client.login(obj)
   }
```

**Comment by mooreds** [comment]
How have you configured CORS in FusionAuth? Here's the docs about it: https://fusionauth.io/docs/v1/tech/reference/cors/

I don't believe tweaking the typescript client is helpful because the issue is with the browser rules. (As a troubleshooting measure, you could always try running the same code in node and ensuring that works, because it won't have CORS limitations.)

**Comment by NickPelov** [comment]
Ok thanks for the information. I'll refer to some documentation on CORS. Thanks for the time and information.

**Comment by mooreds** [comment]
Did the CORS fix work? We can keep this open until you get the issue resolved if you'd like. In other words, feel free to re-open. (Maybe we need to update the documentation, too.)
